### ISSUE TITLE
Allow using HTTP/2 without SSL

### ISSUE THREAD

**Issue by jroper** [akka/akka-http#1849]
While browsers might not speak HTTP/2 without SSL, gRPC clients (eg the official grpc-java client) are easy to configure to not use SSL, and this is very convenient for ad-hoc testing, because it means you don't need to go and create certificates and ensure the client trusts them and so on and so on. This may be particularly useful for people that want to get started quickly, eg if they wanted to validate that Akka HTTP/Akka gRPC is right for them, as it will allow them to defer worrying about configuring SSL till after they've proved that the gRPC side of things works, when they've got more time to burn.

**Comment by jroper** [comment]
There's no spec, there's just different configurable options that the official grpc client/server supports:

https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/NegotiationType.java

**Comment by doofin** [comment]
Still not working without ssl

**Comment by raboof** [comment]
@doofin that is correct: since #1934 we do support 'http2 without TLS with preexisting knowledge', but that is of somewhat limited use (it does not allow using insecure http2 and insecure regular HTTP on the same port) and has not been released yet. h2c support is being worked on in #1966.

**Comment by raboof** [comment]
Nowadays we support both h2c with prior knowledge and h2c via the Upgrade header, so I think it's high time we close this
